pkg/apis/enricher/framework/go/beego_detector.go

```diff
@@ -40,15 +51,16 @@ type ApplicationPropertiesFile struct {
 
 // DoPortsDetection searches for the port in conf/app.conf
 func (b BeegoDetector) DoPortsDetection(component *model.Component, ctx *context.Context) {
-	bytes, err := utils.ReadAnyApplicationFile(component.Path, []model.ApplicationFileInfo{
-		{
-			Dir:  "conf",
-			File: "app.conf",
-		},
-	}, ctx)
+	fileContents, err := utils.GetApplicationFileContents(b.GetApplicationFileInfos(component.Path, ctx))
```

Comment:
Now that all detectors have a ` GetApplicationFileInfos(componentPath string, ctx *context.Context) []model.` method, it seems that you could create an interface with this method, and move the call to `GetApplicationFileInfos` into `utils.GetApplicationFileContents`, something similar to:

```
type FileInfo interface {
  GetApplicationFileInfos(componentPath string, ctx *context.Context) []model
}
```

```
func (b BeegoDetector) DoPortsDetection(component *model.Component, ctx *context.Context) {
  fileContents, err := utils.GetApplicationFileContents(ctx, b, component.Path)
  [...]
```

```
func GetApplicationFileContents(ctx context.Context, obj Fileinfo, path string) ([]string, error) {
  appFileInfos := b.GetApplicationFileInfos(path, ctx)
  [...]
```

Reply:
Isn't it an issue that not some of the `GetApplicationFileInfos` have different functionality? For example, others are just declaring two or three application files while others are generating a list of potential application file names based on language file extension (e.g all files ending in ".go" for golang projects)

Reply:
@feloy as discussed, creating `interfaces` for `detectors` it would be an awesome feature for alizer. As it may introduces a big amount of changes, we can create a separate issue which will only focusing on refactoring